src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Formatting/CSharpFormatter.cs

```diff
@@ -106,9 +106,15 @@ static void ExtractTriviaAnnotations(
             var formattedTriviaList = formattedRoot.GetAnnotatedTrivia(MarkerId);
             foreach (var trivia in formattedTriviaList)
             {
-                // We only expect one annotation because we built the entire trivia with a single annotation.
-                var annotation = trivia.GetAnnotations(MarkerId).Single();
-                if (!int.TryParse(annotation.Data, out var projectedIndex))
+                // We only expect one annotation because we built the entire trivia with a single annotation, but
+                // we need to be defensive here. Annotations are a little hard to work with though, so apologies for
+                // the slightly odd method of validation.
+                using var enumerator = trivia.GetAnnotations(MarkerId).GetEnumerator();
+                enumerator.MoveNext();
+                var annotation = enumerator.Current;
+                // We shouldn't be able to enumerate any more, and we should be able to parse our data out of the annotation.
+                if (enumerator.MoveNext() ||
+                    !int.TryParse(annotation.Data, out var projectedIndex))
```

Comment:
The defensive code seems fine to me, but what are the conditions where multiple annotations are returned?

Reply:
Given this document with source mappings at each `$$`

```
<div>
@{$$
#if DEBUG
$$}

@{$$
#endif
$$}
</div>
```

If DEBUG is not defined, then the whole span between directives is a single "DisabledTextTrivia", but obviously it contains two mappings, so two points to track.